app/models/pillar.rb

```diff
@@ -2,7 +2,7 @@
 # rubocop:disable Metrics/ClassLength
 class Pillar < ApplicationRecord
   validates :pillar, presence: true
-  validates :value, presence: true
+  validates :value, presence: true, format: { with: /\A[\S]+\z/ }
```

Comment:
Wouldn't this regular expression force the value to not contain any spaces at all?

That might be problematic for the `username` and `password` of the cloud configurations.

Reply:
right, i'll update it